### ISSUE TITLE
Question: advantages of data formats

### ISSUE THREAD

**Issue by brockfanning** [datasciencecampus/sdg-indicators#2685]
@dougmet I was hoping to pick your brain about the platform's data formats, as we're looking into revamping data formats in the US platform -- possibly into tidy data, or at least wide data with a meaningful naming convention for columns. I believe the UK data is managed/stored in tidy format, and output in list-form json (cool!). I know when the repo was originally forked the data was in wide format. What were some of the technical and/or administrative advantages of switching to tidy? For example, what kinds of programmatic things did it facilitate or allow? And what kinds of data management practices did it help with? Was there a challenge in getting the data providers to work with tidy data instead of wide?

Lastly - about the list-form json output -- what are the technical advantages of that format? Does having the data in arrays by column help with the visualizations?

**Comment by dougmet** [comment]
Hi @brockfanning,

Yes you're right. The input csvs are in long (tidy) format and we turn this into JSON at build time. For us the format was mostly driven by the disaggregation needs. We have quite long column names and they can combine for several levels so we didn't think it would work in wide format. Long seemed more scalable.

The long format has also made it easier to run automated checks on the data. So for example we know that the `Value` and `Year` columns must be numeric. We've also added other special columns, such as `GeoCode`.

In terms of how data providers have managed. We had surprisingly few problems here. Our ONS colleagues are used to handling data and from the [documentation](https://github.com/datasciencecampus/sdg-indicators/wiki/Raw-data-format) they ran with it. Our main issue has been little things like non-utf8 characters and I was a bit anal about case-sensitive rules opting to keep the data clean at some inconvenience to them. Prose helped, although recently they are circumventing it a bit and opting for more bulk approaches.

The reason for the list-form was data size. After we add geographic data some of our datasets are getting pretty big (>7Mb). Until we have a data service that whole lot has to go over in one request. We found that even after compression the list format saved enough space to be worth it. All things being equal a more standard format (rows) would probably have been chosen.
